### ISSUE TITLE
[BUG] `wash push` to jFrog artifactory 

### ISSUE THREAD

**Issue by ptsgr** [wasmCloud/wasmCloud#3405]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [X] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Describe the bug

When I trying to push any artifacts with size more then `4mb` via `wash`, push command return success code. But when I trying to download it I got an error:
```
Invalid provider archive: numeric field did not have utf-8 text: *** when getting cksum for **********
```

I did my own investigation and came to the conclusion that the problem is in the chunked load.
Becuse artifact layer in registry have same size as last chunk.
<img width="1159" alt="Screenshot 2024-10-22 at 3 48 22 PM" src="https://github.com/user-attachments/assets/3d90130c-f674-4587-b1cb-645dda25f3e0">

Logs from push:
```
DEBUG oci_distribution::client: Pushing chunk start_byte=20971520 end_byte=21628742 blob_data_len=21628743 body_len=657223 location="my.registry.com/external/providers/http-server/blobs/uploads/06325aaf-a022-439e-ad70-c65d5ebba93b" headers={"content-range": "20971520-21628742", "content-length": "657223", "content-type": "application/octet-stream"}
```

With artifacts with size less then `4mb` wash push works fine.

I guess issue related with https://github.com/oras-project/rust-oci-client

### Steps to reproduce

```bash
wash pull ghcr.io/wasmcloud/http-server:0.23.1
wash push my.artifactory.com/providers/http-server:0.23.1 ./http-server.par.gz
```

```bash
wash pull my.artifactory.com/providers/http-server:0.23.1
```

### Expected behavior

Artifact pushed to artifactory with correct layer size.
Artifact can be pulled after successful push.

### Environment

- OS: Linux, MacOS
- Shell: bash, zsh

**Comment by brooksmtownsend** [comment]
I think @joonas has been running point on fixing this upstream in rust-oci-client, once that's ready we should bring it into wash

**Comment by joonas** [comment]
That's right, once https://github.com/oras-project/rust-oci-client/pull/177 lands (and possibly https://github.com/oras-project/rust-oci-client/pull/176, I'm not 100% on how Artifactory behaves in this regard), we should be able to add an option to wash to do a monolithic push instead of chunked.

**Comment by joonas** [comment]
@ptsgr I've https://github.com/wasmCloud/wasmCloud/pull/3461 up for review, the intention is to address this issue with the new feature in `wash` (and updates to the underlying OCI crates).

If you have a chance to pull down my branch and compile `wash` locally off of that branch to try it out, I would be interested to hear if that addresses the problem you're seeing.

**Comment by joonas** [comment]
Since #3461 shipped as part of [`wash v0.37.0`](https://github.com/wasmCloud/wasmCloud/releases/tag/wash-cli-v0.37.0) I'll go ahead and close this out.

Please feel free to re-open the issue if you experience any further issues!
